diff -r ac4cb5848284 sqlserver_ado/compiler.py
--- a/sqlserver_ado/compiler.py	Tue May 06 10:23:54 2014 -0400
+++ b/sqlserver_ado/compiler.py	Wed May 07 09:36:31 2014 +0200
@@ -384,12 +384,17 @@
     pass
 
 class SQLUpdateCompiler(compiler.SQLUpdateCompiler, SQLCompiler):
-    def as_sql(self):
-        sql, params = super(SQLUpdateCompiler, self).as_sql()
-        if sql:
-            # Need the NOCOUNT OFF so UPDATE returns a count, instead of -1
-            sql = 'SET NOCOUNT OFF; {0}; SET NOCOUNT ON'.format(sql)
-        return sql, params
+    def execute_sql(self, result_type):
+        # Hack -- use a straight pymssql cursor to avoid throwing query counts off.
+        # Run the update_only_fields tests if you change this.
+        cursor = self.connection.connection.cursor()
+        try:
+            cursor.execute('SET NOCOUNT OFF')
+            result = super(SQLUpdateCompiler, self).execute_sql(result_type)
+            cursor.execute('SET NOCOUNT ON')
+            return result
+        finally:
+            cursor.close()
 
 class SQLAggregateCompiler(compiler.SQLAggregateCompiler, SQLCompiler):
     def as_sql(self, qn=None):
